### PR TITLE
New sensible defaults

### PR DESCRIPTION
**Pull request by stoprightthere**

### CODE REVIEW

File: geometric_kernels/kernels/matern_kernel.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+
+@dispatch
+def default_feature_map(space: NoncompactSymmetricSpace, *, num, kernel):
+    return random_phase_feature_map_noncompact(space, num)
+
+
+@dispatch
+def default_num(space: Mesh):
+    return min(MaternGeometricKernel._MAX_NUM_EIGENFUNCTIONS, space.num_vertices)
```
Comment (vabor112) [inline geometric_kernels/kernels/matern_kernel.py:33]: Change `_MAX_NUM_EIGENFUNCTIONS` to `_DEFAULT_NUM_EIGENFUNCTIONS`

File: geometric_kernels/kernels/matern_kernel.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+class MaternGeometricKernel:
+    """
+    This class represents a Matérn geometric kernel that "just works".
+
+    Upon creation, this class unpacks into a specific geometric kernel based
+    on the provided Space, and the associated feature map.
+    """
+
+    _MAX_NUM_EIGENFUNCTIONS = 1000
+    _MAX_NUM_LEVELS = 10
```
Comment (vabor112) [inline geometric_kernels/kernels/matern_kernel.py:60]: I suggest going for `100`

File: geometric_kernels/kernels/matern_kernel.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+    def __new__(cls, space: Space, num=None, **kwargs):
+        r"""
+        Construct a kernel and a feature map on `space`.
+
+        :param space: space to construct the kernel on
+        :param num: if provided, controls the "order of approximation"
+            of the kernel. For the discrete spectrum spaces, this means
+            the number of unique eigenvalues that go into the truncated
+            series that define the kernel. For the noncompact symmetric
```
Comment (vabor112) [inline geometric_kernels/kernels/matern_kernel.py:71]: define->defines

File: geometric_kernels/kernels/matern_kernel.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            spaces, this is the number of random phases to construct the
+            kernel.
+        :param **kwargs: any additional keyword arguments to be passed to
+            the kernel (like `key`).
+        """
+
+        # good ole isinstance
```
Comment (vabor112) [inline geometric_kernels/kernels/matern_kernel.py:78]: rm

File: geometric_kernels/kernels/feature_maps.py [hunk 5]
```diff
@@ -4,24 +4,43 @@
...
 
 from geometric_kernels.kernels import MaternKarhunenLoeveKernel
 from geometric_kernels.lab_extras import from_numpy
 from geometric_kernels.sampling.probability_densities import base_density_sample
 from geometric_kernels.spaces import DiscreteSpectrumSpace, NoncompactSymmetricSpace
 
 
-@dispatch
-def deterministic_feature_map(
+def deterministic_feature_map_compact(
```
Comment (vabor112) [inline geometric_kernels/kernels/feature_maps.py:14]: Maybe change `_compact` to `_discrete`, to align with `DiscreteSpectrumSpace`?

File: geometric_kernels/kernels/feature_maps.py [hunk 6]
```diff
@@ -46,17 +65,56 @@ def _map(X: B.Numeric, params, state, **kwargs) -> B.Numeric:
     return _map
 
 
-@dispatch
-def random_phase_feature_map(
+def random_phase_feature_map_compact(
```
Comment (vabor112) [inline geometric_kernels/kernels/feature_maps.py:68]: Maybe change `_compact` to `_discrete`, to align with `DiscreteSpectrumSpace`?

**Review by vabor112 — APPROVED** [review]
Here's some comments. Overall LGTM. **Please, when merging, create an issue asking to align the example notebooks with the new sensible defaults**.

File: geometric_kernels/kernels/matern_kernel.py (hunk 2 above)
Comment (stoprightthere) [reply geometric_kernels/kernels/matern_kernel.py:60]: There's a problem with that: https://github.com/vdutor/SphericalHarmonics/issues/11

File: geometric_kernels/kernels/matern_kernel.py (hunk 3 above)
Comment (stoprightthere) [reply geometric_kernels/kernels/matern_kernel.py:71]: TIL that series is a singular noun

**Pull request merged**